deep_water_level/src/infer.py

```diff
@@ -69,116 +71,128 @@ def run_gradio_app(model, crop_box=None):
 
 def run_dataset_inference(
     model,
-    dataset_dir,
-    annotations_file,
+    dataset_dir: Path,
+    annotations_file: str,
     normalized_output,
     crop_box=None,
     use_water_line=False,
 ):
-    # tensor to string
-    def t2s(t):
-        t = t.squeeze()
-        if len(t.shape) == 0 or t.shape[0] == 1:
-            return f"{t.item():.2f}"
-
-        return "[" + ", ".join([f"{x:.2f}" for x in t]) + "]"
-
-    # Load the dataset
-    dataset = get_data_loader(
-        dataset_dir + "/images",
-        dataset_dir + "/annotations/" + annotations_file,
-        shuffle=False,
-        crop_box=crop_box,
+    # Convert numpy array to string for printing
+    def a2s(a):
+        assert len(a.shape) <= 1
+        if len(a.shape) == 0 or a.shape[0] == 1:
+            return f"{a.item():.4f}"
+        else:
+            return "[" + ", ".join([f"{x:.4f}" for x in a]) + "]"
+
+    transforms = get_transforms(crop_box=crop_box)
+    dataset = WaterDataset(
+        dataset_dir / "annotations" / annotations_file,
+        dataset_dir / "images",
+        transforms=transforms,
         normalize_output=normalized_output,
         use_water_line=use_water_line,
     )
 
-    # Run inference
-    loss = 0
-    n_images = 0
-    outputs = []
-    labeled_depths = []
-
+    # Run inference on each image in the dataset
     data = []
+    for i, (image, depth, filename) in enumerate(dataset):
+        output = run_inference(model, image)
 
-    for i, (images, depths, filenames) in enumerate(dataset):
-        mse = 0
-        for image, depth, filename in zip(images, depths, filenames):
-            output = run_inference(model, image)
-            outputs.append(output)
-            if not use_water_line:
-                labeled_depths.append(depth.item())
-            error = abs(output - depth)
-
-            if VERBOSE:
-                print(f"Filename: {filename}, Infered: {t2s(output)}, Actual: {t2s(depth)}, Error: {t2s(error)}")
-            try:
-                timestamp = filename_to_datetime(filename)
-            except ValueError:
-                # For some synthetic datasets, the filename is not a timestamp
-                timestamp = None
-            data.append(
-                {
-                    "timestamp": timestamp,
-                    "filename": filename,
-                    "predicted": t2s(output),
-                    "actual": t2s(depth),
-                }
-            )
-            mse += error**2
-        mse /= len(images)
-        if VERBOSE:
-            print(f"MSE ({i}): {mse}")
-        loss += mse
-        n_images += len(images)
+        # Convert tensors to 1d numpy arrays
```

Comment:
Converting every thing to numpy arrays instead of tensors here made it easier to store in a dataframe and manipulate later.

Reply:
yey, nice clean up!